src/BackgroundTask/RecurringBackgroundTask.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Abstract class RecurringBackgroundTask.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\BackgroundTask;
+
+/**
+ * Abstract base class for using cron to execute a background task that runs on a schedule.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1
+ * @internal
+ */
+abstract class RecurringBackgroundTask extends CronBasedBackgroundTask {
+
+	/**
+	 * Register the service with the system.
+	 */
+	public function register() {
+		parent::register();
+
+		add_action( 'admin_init', [ $this, 'schedule_event' ] );
+		add_action( $this->get_event_name(), [ $this, 'process' ] );
+	}
+
+	/**
+	 * Schedule the event.
+	 *
+	 * @param mixed[] ...$args Arguments passed to the function from the action hook.
+	 */
+	public function schedule_event( ...$args ) {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
```

Comment:
Why the `manage_options` check? Is this to prevent scheduling for frontend requests. Yeah, we've done that before.

Reply:
Perhaps it would be safer to gate this behind just a user being logged-in:

```suggestion
		if ( ! is_user_logged_in() ) {
			return;
		}
```

Reply:
Yes, this works.